### PR TITLE
Update QR code Library from qrcode to pyQRCode

### PR DESCRIPTION
**Pull request by mostafaradwann**

### CODE REVIEW

File: ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py [hunk 1]
```diff
@@ -340,17 +340,14 @@ def add_integration_log_document(self, zatca_message, integration_status, zatca_
...
-            img_str = base64.b64encode(img_buffer.read()).decode("utf-8")
+            qr = pyqrcode.create(self.qr_code)
+            buffer = BytesIO()
+            qr.png(buffer, scale=7)
+            buffer.seek(0)
+            img_str = base64.b64encode(buffer.getvalue()).decode("utf-8")
             return img_str
         else:
-            return "No QR image"
+            return "No QR Code"
```
Comment (mhaggag) [inline ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py:350]: How is the virtual field intended to be used? If the caller embeds its result directly into an `` element, it won't work when it says "No QR Code". We have a couple of options:
* Return `None` when there's no QR code. The caller has to check for truthiness before embedding in an image (i.e. `if doc.qr_code: do_something`)
* Always return an image, but use a special image to indicate no QR code (e.g. load from assets or render an SVG)

I think the first option's easier, especially if we're going to just embed them in print formats.

**Review by mhaggag — CHANGES_REQUESTED** [review]

**Comment by mhaggag** [comment]
Also, please add the package to `pyproject.toml`. We don't control frappe dependencies, so they can potentially replace the library with something else and break our code if we don't add the dependency ourselves.

File: ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py [hunk 2]
```diff
@@ -340,17 +340,14 @@ def add_integration_log_document(self, zatca_message, integration_status, zatca_
...
-            qr = qrcode.QRCode(version=1, box_size=10, border=5)
-            qr.add_data(self.qr_code)
-            qr.make(fit=True)
-            img = qr.make_image(fill_color="black", back_color="white")
-            img_buffer = BytesIO()
-            img.save(img_buffer, format='PNG')
-            img_buffer.seek(0)
-            img_str = base64.b64encode(img_buffer.read()).decode("utf-8")
+            qr = pyqrcode.create(self.qr_code)
+            buffer = BytesIO()
```
[the remark this answers is not in the document]
Comment (mostafaradwann) [reply ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py]: done

File: ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py [hunk 3]
```diff
@@ -340,17 +340,14 @@ def add_integration_log_document(self, zatca_message, integration_status, zatca_
...
-            img = qr.make_image(fill_color="black", back_color="white")
-            img_buffer = BytesIO()
-            img.save(img_buffer, format='PNG')
-            img_buffer.seek(0)
-            img_str = base64.b64encode(img_buffer.read()).decode("utf-8")
+            qr = pyqrcode.create(self.qr_code)
+            with BytesIO() as buffer:
+                qr.png(buffer, scale=7)
+                buffer.seek(0)
+                img_str = base64.b64encode(buffer.getvalue()).decode("utf-8")
```
Comment (mhaggag) [inline ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py:347]: Please keep the return statement in the with block (indent)

File: ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py [hunk 4]
```diff
@@ -340,17 +340,14 @@ def add_integration_log_document(self, zatca_message, integration_status, zatca_
...
             return img_str
         else:
-            return "No QR image"
+            return None
```
Comment (mhaggag) [inline ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py:350]: We can now invert the if and exit early (i.e. `if not self.qr_code:` and exit the function to unindent the rest)

**Review by mhaggag — APPROVED** [review]

**Comment by mhaggag** [comment]
The print format changes were included by mistake. Please do an interactive rebase and remove them.

**Pull request merged**